packages/vite/src/node/plugins/clientInjections.ts

```diff
@@ -24,20 +24,23 @@ export function clientInjectionsPlugin(config: ResolvedConfig): Plugin {
         const timeout = options.timeout || 30000
         const overlay = options.overlay !== false
         let port
-        if (config.server.middlewareMode) {
-          port = String(
-            (typeof config.server.hmr === 'object' && config.server.hmr.port) ||
-              24678
-          )
-        } else {
-          port = String(options.port || config.server.port!)
-        }
-        let hmrBase = config.base
-        if (options.path) {
-          hmrBase = path.posix.join(hmrBase, options.path)
-        }
-        if (hmrBase !== '/') {
-          port = path.posix.normalize(`${port}${hmrBase}`)
+        if (!options.runtimePort) {
+          if (config.server.middlewareMode) {
+            port = String(
+              (typeof config.server.hmr === 'object' &&
+                config.server.hmr.port) ||
+                24678
+            )
+          } else {
+            port = String(options.port || config.server.port!)
+          }
+          let hmrBase = config.base
+          if (options.path) {
+            hmrBase = path.posix.join(hmrBase, options.path)
+          }
+          if (hmrBase !== '/') {
+            port = path.posix.normalize(`${port}${hmrBase}`)
+          }
         }
```

Comment:
With this logic we're losing appending the `hmrBase` Was that intended?

It's also less readable to move the runtimePort check until later.

This would retain the behaviour:

```ts
let port
if (options.runtimePort != null) {
  port = runtimePort
} else if (config.server.middlewareMode) {
  port = String(
    (typeof config.server.hmr === 'object' && config.server.hmr.port) || 24678,
  )
} else {
  port = String(options.port || config.server.port!)
}

let hmrBase = config.base
if (options.path) {
  hmrBase = path.posix.join(hmrBase, options.path)
}
if (hmrBase !== '/') {
  port = path.posix.normalize(`${port}${hmrBase}`)
}
```

Reply:
Thanks for pointing this out. I will append hmrBase in my next commit.

Reply:
https://github.com/vitejs/vite/blob/07dbae0d92bfa97b993a6fba3e1cf4272b217f1b/packages/vite/src/node/plugins/clientInjections.ts#L46-L49

I have gotten `hmrBase` back.